src/vma/sock/sockinfo_udp.cpp

```diff
@@ -995,10 +995,13 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
 
 					struct ip_mreqn mreqn;
 
+					memset(&mreqn, 0, sizeof(mreqn));
+
 					if (__optlen >= sizeof(struct ip_mreqn)) {
 						mreqn = *(struct ip_mreqn*)__optval;
+					} else if (__optlen == sizeof(struct ip_mreq)) {
+						mreqn.imr_address = ((struct ip_mreq*)__optval)->imr_interface;
```

Comment:
could you double check because I think that `mreqn.imr_address = ((struct ip_mreq*)__optval)->imr_interface;` seems incorrect.
Probably you can use `memcpy(&mreqn, __optval, sizeof(struct ip_mreq));` for case two

Reply:
struct ip_mreq
{
struct    in_addr imr_multiaddr;       /* IP multicast address of group */
**struct    in_addr imr_interface;**         /* local IP address of interface */
};

struct ip_mreqn
{
struct    in_addr imr_multiaddr;       /* IP multicast address of group */
**struct    in_addr imr_address;**         /* local IP address of interface */
int           imr_ifindex;                    /* Interface index */
};

imr_address and imr_interface are the same and the only important address we care about. But your suggestion will work as well.

Reply:
I do not see assignment to `imr_multiaddr`

Reply:
True. But it is not used when handling IP_MULTICAST_IF command. We only care about the interface's address. In other commands (Like IP_ADD_MEMBERSHIP)

Reply:
I prefer to use complete assignment.
In addition. Please consider following if else condition
```
if (__optlen == sizeof(struct ip_mreqn))
else if ((__optlen == sizeof(struct ip_mreq)))
else if (((__optlen == sizeof(struct addrin_t))))
else {
errno=EINVAL;
return with invalid parameter status.
}
```

Reply:
any comments?